JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt

```diff
@@ -39,35 +38,37 @@ fun SelectTopicButton(
     modifier: Modifier = Modifier.None,
     selected: Boolean = false
 ) {
-    Container(modifier = modifier + LayoutSize(36.dp, 36.dp)) {
-        if (selected) {
-            DrawSelectTopicButtonOn()
-        } else {
-            DrawSelectTopicButtonOff()
-        }
+    if (selected) {
+        SelectTopicButtonOn(modifier + LayoutSize(36.dp, 36.dp))
+    } else {
+        SelectTopicButtonOff(modifier + LayoutSize(36.dp, 36.dp))
     }
 }
 
 @Composable
-private fun DrawSelectTopicButtonOn() {
-    DrawShape(
+private fun SelectTopicButtonOn(modifier: Modifier = Modifier.None) {
+    Box(
+        backgroundColor = MaterialTheme.colors().primary,
         shape = CircleShape,
-        color = MaterialTheme.colors().primary
-    )
-    DrawVector(vectorResource(R.drawable.ic_check))
+        modifier = modifier
+    ) {
+        DrawVector(vectorResource(R.drawable.ic_check))
+    }
 }
 
 @Composable
-private fun DrawSelectTopicButtonOff() {
+private fun SelectTopicButtonOff(modifier: Modifier = Modifier.None) {
     val borderColor = MaterialTheme.colors().onSurface.copy(alpha = 0.12f)
-    DrawShape(
+    Box(
+        backgroundColor = borderColor,
         shape = CircleShape,
-        color = borderColor
-    )
-    DrawVector(
-        vectorImage = vectorResource(R.drawable.ic_add),
-        tintColor = MaterialTheme.colors().primary
-    )
+        modifier = modifier
+    ) {
+        DrawVector(
+            vectorImage = vectorResource(R.drawable.ic_add),
+            tintColor = MaterialTheme.colors().primary
```

Comment:
This should probably be `contentColor()`

Reply:
+1 switching.